### PR TITLE
homebrew_cask: Consider greedy option for upgrades too

### PR DESCRIPTION
**Pull request by janernsting**
##### SUMMARY
`brew-cask` accepts `--greedy` for both the upgrade and outdated command (see [their usage reference](https://github.com/Homebrew/homebrew-cask/blob/master/USAGE.md)). So far, only the check for outdated casks considered the Ansible option.

This change aligns Ansible's use of the greedy option.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
homebrew_cask

### CODE REVIEW

**Comment by felixfontein** [comment]
You need a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

**Comment by janernsting** [comment]
Does the add changelog fragment match your needs, @felixfontein?

File: changelogs/fragments/798-homebrew-cask-greedy-for-upgrades-too.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - homebrew_cask - consider `greedy` option for upgrade_all, too (https://github.com/ansible-collections/community.general/pull/798)
```
Comment (felixfontein) [inline changelogs/fragments/798-homebrew-cask-greedy-for-upgrades-too.yaml:2]:
```suggestion
  - homebrew_cask - consider ``greedy`` option for upgrade_all, too (https://github.com/ansible-collections/community.general/pull/798)
```

**Comment by felixfontein** [comment]
@janernsting sorry I missed the update. Looks good, but it uses reStructuredText formatting and not MarkDown, so it should be double backticks.

@Akasurde @danieljaouen @enriclluelles @indrajitr @kyleabenson @martinm82 as maintainers of the module, could you please look at the change itself whether it is good?

File: changelogs/fragments/798-homebrew-cask-greedy-for-upgrades-too.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - homebrew_cask - consider ``greedy`` option for upgrade_all, too (https://github.com/ansible-collections/community.general/pull/798)
```
Comment (felixfontein) [inline changelogs/fragments/798-homebrew-cask-greedy-for-upgrades-too.yaml:2]:
```suggestion
  - homebrew_cask - consider ``greedy`` option for upgrade_all, too (https://github.com/ansible-collections/community.general/pull/798).
```
Sorry, I totally missed that period...

**Comment by janernsting** [comment]
@felixfontein I amended the previous commit to include a trailing period.

**Comment by Andersson007** [comment]
@janernsting thanks for the PR!
@martinm82 could you please take a look?

**Review by martinm82 — APPROVED** [review]

File: plugins/modules/packaging/os/homebrew_cask.py [hunk 3]
```diff
@@ -540,6 +540,9 @@ def _upgrade_all(self):
             [self.brew_path, 'cask', 'upgrade']
         )
 
+        if self.greedy:
```
Comment (martinm82) [inline plugins/modules/packaging/os/homebrew_cask.py:543]: Please update as well the documentation (`DOCUMENTATION`) of the `greedy` attribute . Right now it mentions only that it is used for outdated command

**Review by martinm82 — CHANGES_REQUESTED** [review]

**Comment by felixfontein** [comment]
@janernsting can you please take a look at @martinm82's review comment?

**Comment by janernsting** [comment]
Thank you for your heads-up, @felixfontein. I am likely to get this done with next week or in between Xmas and New Years eve.

**Comment by felixfontein** [comment]
@janernsting are you still working on this?

needs_info

**Comment by felixfontein** [comment]
Closing since there was no more reaction. @janernsting if you're interested again in working on this, please ping me. If someone else is interested in continuing this PR, please also write it here so we can coordinate.

**Pull request closed without merging**